### ISSUE TITLE
Portainer Widget HTTPS

### ISSUE THREAD

**Issue by Samurai1201** [benphelps/homepage#417]
**Describe the bug**
Portainer Widget doesnt work as intended, either due to https, or not accepting my env input

**Expected behavior**
Portainer should display as a widget

**Additional context**
- Happens with both "latest" image
- Docker version 20.10.19
- Deployment method: Docker-compose/portainer stack

Config lines:
```
- Portainer:
            icon: /icons/portainer.png
            description: Docker verwalten
            href: https:// -IP- :9443
            widget:
              type: portainer
              url: https:// -IP- :9443
              env: 2
              key: - Redacted -
```

Error message:

```
Something went wrong.
TypeError: l.filter is not a function

c@http://172.28.110.158:3000/_next/static/chunks/8019.fdf32b58caf74612.js:1:880
5589/y/h<@http://172.28.110.158:3000/_next/static/chunks/4931-ff1596ae4f86f149.js:1:14837
n@http://172.28.110.158:3000/_next/static/chunks/pages/index-30b83ae3c35e861e.js:1:2657
G@http://172.28.110.158:3000/_next/static/chunks/pages/index-30b83ae3c35e861e.js:1:7391
div
li
Z@http://172.28.110.158:3000/_next/static/chunks/pages/index-30b83ae3c35e861e.js:1:8521
ul
C@http://172.28.110.158:3000/_next/static/chunks/pages/index-30b83ae3c35e861e.js:1:11086
div
I@http://172.28.110.158:3000/_next/static/chunks/pages/index-30b83ae3c35e861e.js:1:11373
div
div
K@http://172.28.110.158:3000/_next/static/chunks/pages/index-30b83ae3c35e861e.js:1:19712
n@http://172.28.110.158:3000/_next/static/chunks/pages/index-30b83ae3c35e861e.js:1:2657
6747/G<@http://172.28.110.158:3000/_next/static/chunks/pages/_app-fc77031352f30888.js:1:82737
V@http://172.28.110.158:3000/_next/static/chunks/pages/index-30b83ae3c35e861e.js:1:17771
div
div
Y@http://172.28.110.158:3000/_next/static/chunks/pages/index-30b83ae3c35e861e.js:1:21904
a@http://172.28.110.158:3000/_next/static/chunks/pages/_app-fc77031352f30888.js:1:73304
s@http://172.28.110.158:3000/_next/static/chunks/pages/_app-fc77031352f30888.js:1:73861
u@http://172.28.110.158:3000/_next/static/chunks/pages/_app-fc77031352f30888.js:1:72736
6747/G<@http://172.28.110.158:3000/_next/static/chunks/pages/_app-fc77031352f30888.js:1:82737
6207/t.default<@http://172.28.110.158:3000/_next/static/chunks/pages/_app-fc77031352f30888.js:1:72186
F@http://172.28.110.158:3000/_next/static/chunks/pages/_app-fc77031352f30888.js:1:7870
n@http://172.28.110.158:3000/_next/static/chunks/pages/_app-fc77031352f30888.js:1:67946
r@http://172.28.110.158:3000/_next/static/chunks/main-d2b206db40ae9853.js:1:6949
Y@http://172.28.110.158:3000/_next/static/chunks/main-d2b206db40ae9853.js:1:9188
ie@http://172.28.110.158:3000/_next/static/chunks/main-d2b206db40ae9853.js:1:11828
```

**Comment by Samurai1201** [comment]
```
[2022-10-20T17:08:56.920Z] error: can not read a block mapping entry; a multiline key may not be an implicit key (40:20)

 37 |                 type: portainer
 38 |                 url: https:// -IP- :9443
 39 |                 env: 2
 40 |                 key: -Redacted- ...
-------------------------^
```

**Comment by Samurai1201** [comment]
Thanks for the tip with the formatter. Now my code looks way better, but it didnt fix my problem.
Turns out the log message was from a previous misconfiguration and not related to this problem
Sorry

**Comment by shamoon** [comment]
What are the results of `curl -L -H 'X-Api-Key:YOURKEY' 'https://YOURPORTAINERURL/api/endpoints/2/docker/containers/json'` ?

**Comment by Samurai1201** [comment]
curl: (60) SSL certificate problem: self signed certificate
Thats what I already expected, but I dont know how to bypass in hompage

**Comment by benphelps** [comment]
Homepage should handle/ignore the self signed cert just fine.  You can try 

`curl -k -L -H 'X-Api-Key:YOURKEY' 'https://YOURPORTAINERURL/api/endpoints/2/docker/containers/json'`

to have curl do the same.

**Comment by Samurai1201** [comment]
curl: (3) URL using bad/illegal format or missing URL

Now comes up as an additional error message.
It didnt mention this earlier, I am using portainer-ee instead of portainer-ce

**Comment by Samurai1201** [comment]
`curl -L -H -k 'X-Api-Key: - Redacted - ' 'https://192.168.178.77:9443/api/endpoints/2/docker/containers/json'` 
Just to clarify, this is the command I used (with the key being redacted for obvious reasons)

**Comment by benphelps** [comment]
The position of the -k argument matters.

**Comment by shamoon** [comment]
OP please re-open if theres truly a bug here

**Comment by YujiTFD** [comment]
Greetings.

Sorry for re-opening closed ticket, but I might have some insight on the problem.
Researching a similar Portainer widget issue I got following reply for **curl** request:

`{"message":"Permission denied to access environment","details":"Access denied to environment"}`

I granted new user with admin rights and widget worked, although documentation says nothing regarding privileges, while user actually do require at least Helpdesk level privileges for certain environment, Read-only and Standard user won't cut.
Such kind of access seems excessive for API read requests tbh, should it really work this way?
